src/librustc/infer/region_inference/mod.rs

```diff
@@ -1543,7 +1543,7 @@ impl<'a, 'gcx, 'tcx> GenericKind<'tcx> {
         match *self {
             GenericKind::Param(ref p) => p.to_ty(tcx),
             GenericKind::Projection(ref p) => tcx.mk_projection(
-                p.trait_ref.clone(), p.item_name(tcx)),
+                p.trait_ref(tcx).clone(), p.item_name(tcx)),
```

Comment:
Ok. Did the easy part of this - some call sites remain. I'll take another look at those. For example, what am I doing below? Refactor `ast_path_to_mono_trait_ref`?

```rust
        let trait_ref = self.ast_path_to_mono_trait_ref(span,
                                                        trait_def_id,
                                                        self_ty,
                                                        trait_segment);

        debug!("qpath_to_ty: trait_ref={:?}", trait_ref);

        self.normalize_ty(span, tcx.mk_deprecated_projection(trait_ref, item_segment.name))
```